enginetest/queries/tpch_plans.go

```diff
@@ -2930,7 +2930,7 @@ order by
 			"                 ├─ columns: [lineitem_1.l_orderkey:1!null]\n" +
 			"                 └─ Having\n" +
 			"                     ├─ GreaterThan\n" +
-			"                     │   ├─ sum(lineitem.l_quantity):50!null\n" +
+			"                     │   ├─ sum(lineitem.l_quantity):0!null\n" +
```

Comment:
The motivating bug was that HAVING expressions resolve the first instance of a definition. Looks like there is an alias on this aggregate, and we now use the id to match the first instance.

Reply:
In the general case we try to match the latest definition of a variable, because that one is always in the closest scope. In this cases there is some weirdness where we're considering an alias and its getField child from an aggregate into a HAVING as equivalent. James needs this PR to keep making progress here.